presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/rest-openapi.yml

```diff
@@ -0,0 +1,170 @@
+openapi: 3.0.0
+info:
+  title: Presto FunctionNameSpace Rest API
+  description: API for retrieving functions in Presto.
+  version: "1"
+servers:
+  - url: http://localhost:8080
+    description: Presto endpoint when running locally
+# http://127.0.0.1:7777/v1/function/array_constructor?returnType=array(unknown)&numArgs=0
+paths:
+  /v1/function/{functionName}:
+    post:
+      summary: Retrieve value from function
+      parameters:
+        - in: path
+          name: functionName
+          schema:
+            $ref: '#/components/schemas/QualifiedObjectName/properties/objectName'
+          required: true
+          description: The return type of the function of the user to get
+        - in: query
+          name: returnType
+          schema:
+            $ref: '#/components/schemas/TypeSignature/properties/base'
+          required: true
+          description: The return type of the function
+        - in: query
+          name: numArgs
+          schema:
+            type: integer
+          required: true
+          description: The numbers of arguments passed in
+        - in: query
+          name: argType
+          schema:
+            type: object
+            # $ref: '#/components/schemas/TypeSignature/parameters'
+            additionalProperties:
+              type: string
+              pattern: '^[A-Za-z][A-Za-z0-9]*$'
+            example:
+              argType1: string
+              argType2: double
+          description: The type for each argument passed in
+      requestBody:
+        required: true
+        content:
+          text/plain; charset=utf-8:
+            schema:
+              $ref: '#/components/schemas/SerializedPage'
+              type: object
+              properties:
+                username:
+                  type: string
+                  format: binary
+      responses:
+        '200':
+          description: function value output
+          content:
+            text/plain; charset=utf-8:
+              schema:
+                $ref: '#/components/schemas/SerializedPageBinary'
+components:
+  schemas:
+    QualifiedObjectName:
+      type: object
+      properties:
+        catalogName:
+          type: string
+          description: The name of the catalog
+        schemaName:
+          type: string
+          description: The name of the Schema
+        objectName:
+          type: string
+          description: The name of the function
+    TypeSignature:
```

Comment:
TypeSignatures are serialized as simple strings.  I would just make this a string, and link to the code for an example of how it's serialized.